builtin/common/misc_helpers.lua

```diff
@@ -159,6 +159,56 @@ function dump(o, indent, nested, level)
 end
 
 --------------------------------------------------------------------------------
+-- Allow indexing strings
+
+getmetatable"".__index = function(str, index)
+	local itype = type(index)
+	if itype == "string" then
+		return string[index] or str:find(index)
```

Comment:
I don't think you will get around leaving any string index for string.
What about:
```lua
("rep"):rep(3)
```